packages/gatsby-plugin-cms/src/utils/fetch.ts

```diff
@@ -1,10 +1,20 @@
 import unfetch from 'isomorphic-unfetch'
 import retry from 'fetch-retry'
 
-const fetch = (input: RequestInfo, init?: RequestInit) =>
+export const fetch = (input: RequestInfo, init?: RequestInit) =>
   retry(unfetch, {
     retries: 3,
     retryDelay: 500,
   })(input, init)
 
-export default fetch
+export const fetchAPI = async <T>(input: RequestInfo, init?: RequestInit) => {
+  const response = await fetch(input, init)
+
+  if (response.ok) {
+    return response.json() as Promise<T>
+  }
+
+  console.error(await response.text())
+
+  throw new Error(`Error while fetching ${input}`)
```

Comment:
I suggest putting this code inside an if branch and removing the response.json() from the branch. Response returning "ok" is the "happy path," so put it as most left as possible.